tests/cross_functional/system_test/test_full_cluster_health.py

```
@@ -115,102 +115,203 @@ def delete_pods(self):
         Wrapper function for cluster health check
 
+        Args:
+            ceph_recovery_timeout (int): Time to wait for Ceph to recover (default: 600s/10min)
+
         Returns:
-            bool: True if ALL checks passed, False otherwise
+            bool: True if ALL checks passed (Ceph healthy AND all pods running), False otherwise
         """
-        return self.ceph_not_health_error() and pod.wait_for_pods_to_be_running()
+        logger.debug(
+            f"Starting cluster health check with ceph_recovery_timeout={ceph_recovery_timeout}s"
```

Ruff (0.15.20)

[warning] Logging statement uses f-string

(G004)